### PR TITLE
[PUBSUB-33] Remove websocket connection

### PR DESCRIPTION
**Pull request by RobCatch**
https://jira.appcelerator.org/browse/PUBSUB-33

This PR removes the websocket connection from the client, with all events published as a POST to the pubsub server. This makes the client capable of publishing only.

## Verification
Make sure events are still queued and published correctly.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -185,52 +185,9 @@ function PubSubClient(opts) {
...
-					key: opts.key,
-					url: self.url,
-					fingerprint: fingerprint
-				});
-			} catch(e) {
-				self.emit('error', e);
-			}
-		});
 	}
 	this.queueSendDelay = opts.queueSendDelay === undefined ? 10 : Math.max(1, +opts.queueSendDelay);
```
Comment (pkclark) [inline lib/index.js]: @RobCatch let's also remove queuing and delayed send (and related conf and methods). Prefer to just fire immediately with retry.

File: lib/index.js [hunk 2]
```diff
@@ -519,24 +454,11 @@ function toSubArray(value) {
...
-	debug('deliver', event);
-	Object.keys(this.eventSubscriptions).forEach(function (subid) {
-		var pattern = this.eventSubscriptions[subid];
-		if (pattern.test(event.event)) {
-			this.emit(subid, event);
-		}
-	}.bind(this));
-};
-
 PubSubClient.prototype.close = function () {
```
Comment (pkclark) [inline lib/index.js]: @RobCatch should this still be necessary?

**Comment by RobCatch** [comment]
@pkclark Updated. Seems like the close function isn't really all that necessary without a queue.

**Comment by pkclark** [comment]
@RobCatch needs a major version bump.

**Comment by pkclark** [comment]
@RobCatch  are you able to address the linting founds and failing test in the node 6 travis pass?

**Pull request merged**